packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -433,133 +669,242 @@ fn array_to_user_warnings(array: mv8::Value<'_>) -> Vec<UserWarning> {
     }]
 }
 
-fn get_js_error(_mv8: &MiniV8, return_val: &mv8::Object<'_>) -> Option<Error> {
-    if let Ok(errors) = return_val.get("user_errors") {
-        if !matches!(errors, mv8::Value::Undefined) && !matches!(errors, mv8::Value::Null) {
-            let errors = array_to_user_errors(errors);
+fn get_js_error<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    return_val: v8::Local<'s, v8::Object>,
+) -> Option<JSRunnerError> {
+    tracing::debug!("Collecting JS errors");
+    let user_errors = if let Some(user_errors) = v8::String::new(scope, "user_errors") {
```

Comment:
Can we use `new_js_string` instead?

Reply:
I don't think we can.
When I added `new_js_string` I replaced every string creation except in two functions.
These functions don't return `Result` but `Option<Result>`.

`new_js_string` returns a `Result` and I don't know any easy conversion to be able to use `?`.
It's either discarding the value or the error...